src/adapter/stackTrace.ts

```diff
@@ -137,14 +137,23 @@ export class StackTrace {
   }
 
   async formatAsNative(): Promise<string> {
-    const stackFrames = await this.loadFrames(50);
-    const promises = stackFrames.map(frame => frame.formatAsNative());
-    return (await Promise.all(promises)).join('\n') + '\n';
+    return await this.formatWithMapper(frame => frame.formatAsNative());
   }
 
   async format(): Promise<string> {
-    const stackFrames = await this.loadFrames(50);
-    const promises = stackFrames.map(frame => frame.format());
+    return await this.formatWithMapper(frame => frame.format());
+  }
+
+  private async formatWithMapper(mapper: (frame: StackFrame) => Promise<string>): Promise<string> {
+    let stackFrames = await this.loadFrames(50);
+    // REPL may call back into itself; slice at the highest REPL eval in the call chain.
+    for (let i = stackFrames.length - 1; i >= 0; i--) {
```

Comment:
You can also use `findLastIndex`

Reply:
Unfortunately not, for some reason `findIndex` can take a predicate, but `findLastIndex` cannot. (???)